kythe/go/indexer/emit.go

```diff
@@ -983,6 +984,29 @@ func (e *emitter) writeDoc(comments *ast.CommentGroup, target *spb.VName) {
 	e.writeFact(docNode, facts.NodeKind, nodes.Doc)
 	e.writeFact(docNode, facts.Text, strings.Join(lines, "\n"))
 	e.writeEdge(docNode, target, edges.Documents)
+	e.emitDeprecation(target, lines)
+}
+
+// emitDeprecation emits a deprecated fact for the specified target if the
+// comment lines indicate it is deprecated per https://github.com/golang/go/wiki/Deprecated
+func (e *emitter) emitDeprecation(target *spb.VName, lines []string) {
+	var deprecated bool
+	var depMsg string
+	for _, line := range lines {
+		if !deprecated {
+			if msg := strings.TrimPrefix(line, "Deprecated: "); msg != line {
+				depMsg = msg
+				deprecated = true
+			}
+		} else if line == "" {
+			break
+		} else {
+			depMsg += " " + line
```

Comment:
I suggest: https://play.golang.org/p/HPtp-YVDKrF

Key points: Although we're not worried about memory use here, this simplifies both the concatenation issue and the correct handling of spaces. It also cleans up extra whitespace, and handles "morally empty" lines that have only whitespace on them.

Reply:
I actually have just finished a similar change.